session.go

```diff
@@ -152,6 +152,9 @@ func (sess *session) handleRequests(reqs <-chan *gossh.Request) {
 			var kv = struct{ Key, Value string }{}
 			gossh.Unmarshal(req.Payload, &kv)
 			sess.env = append(sess.env, fmt.Sprintf("%s=%s", kv.Key, kv.Value))
+			if req.WantReply {
+				req.Reply(true, nil)
+			}
 		case "pty-req":
```

Comment:
Or perhaps once `pty` is set, it ignores the rest. I can imagine the callback denying for some reason and then it fixing and trying again. No idea if that's a realistic scenario with known SSH clients, but it makes sense implementation wise and resolves this issue.